grakn/service/Session/util/ResponseReader.py

```diff
@@ -127,19 +127,27 @@ def create_explanation(tx_service, grpc_explanation_res):
         native_list_of_concept_maps = []
         for grpc_concept_map in grpc_list_of_concept_maps:
             native_list_of_concept_maps.append(AnswerConverter._create_concept_map(tx_service, grpc_concept_map))
-        return Explanation(native_list_of_concept_maps)
+        rule = ConceptFactory.create_local_concept(grpc_explanation_res.rule)
+        if len(rule.id) == 0:
```

Comment:
@flyingsilverfin 

```suggestion
        if not rule.id:
```

Reply:
i like the explicit length based version as it doesn't require any specialised python knowledge about strings and not behaviour, easy to read for people swapping between langauges a lot :)